script.js

```diff
@@ -53,81 +79,247 @@ my_widget_script =
+              amount.val((moles.val() * fw.val()).toFixed(nFixed))
+            }
           }
         });
-        $('#the_form input[name^=equivalents]').on('keyup change', function() {
+
+        /*
+			Moles Field On Blur Handler
+		*/
+        $('#the_form input[name^=moles]').on('blur', function() {
+          console.log('In moles blur handler')
           var tr = $(this).closest('tr');
+          current_class = tr.attr('class')
+          var compare_class_id = current_class.localeCompare('initialRow');
+          var fw = $('input[name^=fw]', tr);
           var amount = $('input[name^=amount]', tr);
+          var moles = $('input[name^=moles]', tr);
+
+          if (fw.val() && moles.val() && !amount.val()) {
+            console.log('fw field is not empty')
+            amount.val((moles.val() * fw.val()).toFixed(nFixed))
+          }
+
+          // Enable rest of the records of the experiment; if there moles and fw value exist for row
+          if (moles.val() && fw.val() && amount.val && (compare_class_id == 0)) {
+            my_widget_script.enable_all_records()
+          }
+        });
+
+        /*
+			Moles Field On Change Handler
+		*/
+        $('#the_form input[name^=moles]').on('change', function() {
+          var tr = $(this).closest('tr');
+          console.log('In moles change handler'+ tr.attr('class'))
+          current_class = tr.attr('class')
+          var compare_class_id = current_class.localeCompare('initialRow');
+
+          if (compare_class_id == 0) {
+            console.log('Row 1 moles has changed, change row 1 amount and change amount, moles for rest of rows')
+            var amount = $('input[name^=amount]', tr);
+            var fw = $('input[name^=fw]', tr);
+            var moles = $('input[name^=moles]', tr);
+            if (moles.val() && fw.val()) {
+              amount.val((moles.val() * fw.val()).toFixed(nFixed))
+            }
+
+            // Change moles and amount for all enabled rows where value exists
+            my_widget_script.change_record_value()
+          }
+        });
+
+        /*
+			Equivalence Field On Change Handler
+		*/
+        $('#the_form input[name^=equivalents]').on('change', function() {
+          console.log('In equivalents change handler')
+          var tr = $(this).closest('tr');
           var fw = $('input[name^=fw]', tr);
+          var equiv = $('input[name^=equivalents]', tr);
+          var amount = $('input[name^=amount]', tr);
           var moles = $('input[name^=moles]', tr);
-          var moles1 = $('#the_form input[name=moles1_formula]');
+          var moles1 = $('#the_form input[name=moles1_number]');
+          if (fw.val() && moles1.val() && equiv.val()) {
+            moles.val((moles1.val() * equiv.val()).toFixed(nFixed));
+            amount.val((moles.val() * fw.val()).toFixed(nFixed))
+          }
+        });
+
+        /*
+			Equivalence Field On Blur Handler
```

Comment:
This looks very similar to the "Equivalence Field On Change Handler". If they are different there should be comments to the effect. If they are not different or are mostly the same, there should be minimal repeated code.